src/api/code-challenge/content-types/code-challenge/test-evaluator.ts

```diff
@@ -0,0 +1,279 @@
+  } finally {
+    console.log("<metaCaseCode ERROR>", visibleMetaError);
+    console.log("<existsMetaError>", existsMetaError);
+  }
+
+  // Delete Later. Will almost always throw error.
+  // try {
+  //   const internalCheck = eval(internalTestCode);
+  //   console.log("<internalCheck>", internalCheck);
+  // } catch (err) {
+  //   console.log("<internalTestCode ERROR>", err);
+  //   // almost every internalTest will be an Error, partial code
+  // }
+  let evalErrorContainer = null;
+  try {
+    // @ts-expect-error will fix later
+    const context = getEvaluationContext(formattedCode, logs);
+
+    // Error: Max Call Stack Exceeded! Why...? Override Console?
+    // console.log("CONTEXT: ", JSON.stringify(context));
+
+    result = evaluateWithContext(
+      `${formattedCode};
+      ${internalTestCode};`,
+      context
+    );
+
+    //
+    // if (
+    //   typeof result === "boolean" &&
+    //   result !== userShouldPass &&
+    //   result === true
+    // ) {
+    //   descriptionMessage = `ERROR: metaTest ${metaTestId} & internalTest ${internalTest.id} are UNEXPECTEDLY ${result}`;
+    //   userPassed = false;
+    //   evaluationError = { message: null, stack: null };
+    // }
+
+    // AKA if result is correct as expected
+
+    // if Error, catch is run...
+  } catch (err) {
+    // if (typeof result !== "boolean") {
+    //   userPassed = false;
+    //   descriptionMessage = `ERROR: FAILED metaTest ${metaTestId} & internalTest ${
+    //     internalTest.id
+    //   } are type: ${(typeof result).toUpperCase()}, should be BOOLEAN`;
+    // }
+    // if Code fails to compile in try block, this can be normal for some EXPECTED negatives
+
+    evalErrorContainer = err;
+    // descriptionMessage = `CATCH BLOCK ERROR: TEST EVALUATOR, result = evaluateWithContext`;
+    // userPassed = false;
+    // evaluationError = {
+    //   message: `${err.name}: ${err.message}`,
+    //   stack: err.stack,
+    // };
+  } finally {
+    // console.log("---expectPasses: ", expectPasses);
+    console.log("\n<FINALLY RESULT>", result);
+    console.log("<FINAL TYPE>", typeof result, "\n");
+    console.log("<userShouldPass>", userShouldPass);
+    console.log("<metaTestId>", metaTestId);
+    console.log("<internalTest.id>", internalTest.id);
+    console.log("<metaLabel>", metaLabel);
+    console.log("<evalErrorContainer>", evalErrorContainer);
+
+    // Error on vsCode, but ok in run...??? Because object isn't fetched til RUNTIME
+    // console.log("<testLabel>", internalTest.label, "\n");
+
+    // console.log("<evaluationError>", evaluationError);
+
+    const setShortError = () => {
+      evaluationError = evalErrorContainer;
+      evaluationError.stack = evaluationError.stack.substring(0, 200);
+    };
+
+    const setNullError = () => {
+      evaluationError = {
+        message: null,
+        stack: null,
+      };
+    };
+
+    switch (true) {
+      case typeof result !== "boolean" &&
```

Comment:
I know this was an attempt to clean things up but it's still not clean. You have to extract this logic to its own function and then you can just return. And call that function inside of this function and use the result

Reply:
Ok I moved the switch case logic to a separate function called "determineReturnObject()".

Let me know if you have suggestions for the name or implementation here. 

I am trying to preserve the original logic of this section (modifying, then returning an object), but not feeling confident if this is the best way to implement. Should I be experimenting more here?